src/utils/Snippets.ts

```diff
@@ -0,0 +1,101 @@
+  const headers: Header[] = []
+
+  if (document.components?.securitySchemes) {
+    Object.values(document.components?.securitySchemes).forEach(s => {
+      const scheme = deRef(s, document)
+      if ("in" in scheme && scheme.in === "header") {
+        headers.push({name: scheme.name, value: scheme.type})
+      }
+    });
+  }
+  params.forEach(param => {
+    if (param.in === "header" && param.required) {
+      let val = param.name
+      if (param.schema && "type" in param.schema) {
+        val = param.schema.type as string
+      }
+      headers.push({name: param.name, value: val})
+    }
+  })
+
+  // setting 'Content-Type' 'application/json' as a default header
+  const hasContentType = headers.some(header => header.name === 'Content-Type');
+  if (!hasContentType) {
+    headers.push({ name: 'Content-Type', value: 'application/json' });
+  }
+
+  return headers
+}
+
+const getQueryParams = (params: DeRefResponse<OpenAPIV3.ParameterObject>[], document: OpenAPIV3.Document): QueryString[] => {
+  const queryParams = params.filter(param => param.in === 'query' && param.required)
+
+  const paramsWithValue: QueryString[] = queryParams.map(param => {
+    const paramName = param.name
+
+    if (param.example) {
+      return {name: paramName, value: param.example}
+    }
+
+    const paramSchema = param.schema ? deRef(param.schema, document) : undefined
+    if (paramSchema && paramSchema.default) {
+      return {name: paramName, value: paramSchema.default}
+    }
+
+    if (param.examples) {
+      const example = Object.values(param.examples)[0]
+      if ('value' in example) {
+        return {name: paramName, value: example.value}
+      }
+    }
+
+    if (paramSchema && paramSchema.enum) {
+      const enums = paramSchema.enum
+      if (enums.length > 0) {
+        return {name: paramName, value: enums[0]}
+      }
+    }
+
+    return {name: paramName, value: ''}
+  })
+
+  return paramsWithValue
+}
+
+const getPostData = (requestBody: OpenAPIV3.ReferenceObject | OpenAPIV3.RequestBodyObject | undefined, document: OpenAPIV3.Document): PostData | undefined => {
+  if (!requestBody) {
+    return undefined
+  }
+
+  const deRefResponse = deRef(requestBody, document);
+  if (!deRefResponse.content || !(deRefResponse.content && deRefResponse.content['application/json']?.schema)) {
+    return undefined
+  }
+
+  const jsonSchema = recursiveDeRef(deRefResponse.content['application/json'].schema, document);
+  const requestBodyText = JSON.stringify(mock(jsonSchema), undefined, 2);
+
+  return {mimeType: 'application/json', text: requestBodyText}
+}
+
+export const buildCodeSampleData = (verb: string, path: string, params: DeRefResponse<OpenAPIV3.ParameterObject>[], requestBody: OpenAPIV3.ReferenceObject | OpenAPIV3.RequestBodyObject | undefined, document: OpenAPIV3.Document): RequestFormat => {
+  return ({
+    method: verb.toUpperCase(),
+    url: "http://example.com"+path,
```

Comment:
Could we use a relative URL instead? 
We also have the `servers.url` but not everyone might be doing the same - we could read the specs and try to get the teams to use it in an uniform way:

![image](https://user-images.githubusercontent.com/3845764/229815602-b4784930-9737-48fc-9ee7-ff4079d77987.png)

Reply:
@josejulio That was the first thing I tried. Unfortunately, the library only allows valid url for that field. If I use just the relative path, the code samples component does not render at all :disappointed:

Reply:
Ok  - we will try to improve it later